src/server/system_services/replication_store.js

```diff
@@ -71,27 +77,50 @@ class ReplicationStore {
         return ans;
     }
 
-    async update_replication_status_by_id(_id, rule_id, status) {
-        dbg.log0('update_replication_status_by_id: ', _id, rule_id, status);
+    // TODO: revert this function and use the previous impl instead of the current one
+    // before doing it we need to add other ops when converting $set with .$. in postgres client
+    async update_replication_status_by_id(_id, rule_id, rule_status, replication_status) {
+        dbg.log0('update_replication_status_by_id: ', _id, rule_id, rule_status, replication_status);
         const parsed_id = db_client.instance().parse_object_id(_id);
-        const find = { _id: parsed_id, 'rules.rule_id': rule_id };
-        const update = { $set: { 'rules.$.status': status } };
-        const options = { returnOriginal: false };
-        dbg.log1('update_replication_status_by_id: ', find, update, options);
-        const ans = await this._replicationconfigs.findOneAndUpdate(find, update, options);
-        dbg.log1('update_replication_status_by_id: ans', ans);
-        this._replicationconfigs.validate(ans.value, 'warn');
-        return ans;
+
+        const bulk = this._replicationconfigs.initializeOrderedBulkOp();
+        bulk.find({ _id: parsed_id, 'rules.rule_id': rule_id })
```

Comment:
Convert these two queries to a single query instead of doing bulk operation / transaction